<a href="https://colab.research.google.com/github/ParkerCircus/Number-Analyzer/blob/Data-Analytics/Student_Grades.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
import numpy as np
import statistics as stat

In [2]:
math = [11, 12, 11, 7, 9.5, 15, 18]
science = [9, 11, 18, 15.5, 12, 18.5, 15.5]
reading = [16, 8.5, 11.5, 11, 10.5, 7, 11]
history = [13.5, 10, 9, 14, 14, 12, 7.5]

In [3]:
print(np.mean(math), np.mean(science), np.mean(reading), np.mean(history))

11.928571428571429 14.214285714285714 10.785714285714286 11.428571428571429


In [4]:
np.median(math)

11.0

In [5]:
stat.mode(history)

14

In [6]:
print(stat.correlation(math, science), stat.correlation(math, reading), stat.correlation(math, history))
print(stat.correlation(science, reading), stat.correlation(science, history), stat.correlation(reading, history))

0.2694940962141907 -0.2685618890579738 -0.7183583464149537
-0.5276058297032529 -0.358688155652953 0.2067940426483234


Math and history have the strongest correlation of ~ -0.718

In [33]:
test_results = [24, 5, 15, 60, 54, 82, 99, 80, 70, 98, 93, 60, 33, 22, 65, 61, 51, 58, 83, 86, 42, 67, 60]

In [46]:
def desc_stats():
  print(f"Mean: {np.mean(test_results)} Median: {np.median(test_results)} Mode: {stat.mode(test_results)} "
  f"Minimum: {min(test_results)} Maximum: {max(test_results)} Range: {max(test_results) - min(test_results)} "
  f"Variance: {stat.variance(test_results)} Standard deviation: {stat.stdev(test_results)})")

In [47]:
desc_stats()

Mean: 59.47826086956522 Median: 60.0 Mode: 60 Minimum: 5 Maximum: 99 Range: 94 Variance: 694.3517786561265 Standard deviation: 26.350555566365703)


I would not curve the grades. With a median of 60, half the class had a 60 or worse, which is not passing. With a high standard deviation of over 26, this means the test scores were spread out by over two letter grades if selecting a random score out of the data set. This shows a high degree of failure on the test, but also indicates several students did very well showing the test was not mistakeningly over difficult.